src/github.com/mozilla-services/pushgo/simplepush/net.go

```diff
@@ -78,39 +91,50 @@ func ParseRetryAfter(header string) (d time.Duration, ok bool) {
 	return 0, false
 }
 
-// TooBusyError is a temporary error returned when too many simultaneous
-// connections are open. The server will sleep before accepting new
-// connections.
-type TooBusyError struct{}
+type ListenerError struct {
+	Message     string
+	IsTemporary bool
+}
+
+func (err *ListenerError) Error() string   { return err.Message }
+func (err *ListenerError) Timeout() bool   { return false }
+func (err *ListenerError) Temporary() bool { return err.IsTemporary }
 
-func (err TooBusyError) Error() string   { return "Too many requests" }
-func (err TooBusyError) Timeout() bool   { return false }
-func (err TooBusyError) Temporary() bool { return true }
+var (
+	// errTooBusy is a temporary error returned when too many simultaneous
+	// connections are open. The server will sleep before accepting new
+	// connections.
+	errTooBusy = &ListenerError{"Too many requests", true}
 
-var errTooBusy = TooBusyError{}
+	// errClosed is returned when the listener is closed.
+	errClosed = &ListenerError{"Listener closed", false}
+)
 
-// LimitConn decrements the active connection count for closed connections.
-type LimitConn struct {
+// limitConn decrements the active connection count for closed connections.
+type limitConn struct {
 	net.Conn
 	removeOnce sync.Once
 	removeConn func()
 }
 
 // Close implements net.Conn.Close.
-func (c *LimitConn) Close() error {
+func (c *limitConn) Close() error {
 	err := c.Conn.Close()
 	c.removeOnce.Do(c.removeConn)
 	return err
 }
 
+// isTerminalState indicates whether state is a terminal connection state.
```

Comment:
Description is redundant... maybe better to indicate what 'terminal' means in this context? It's dead? It should be terminated? etc.